ocrd_anybaseocr/cli/ocrd_anybaseocr_dewarp.py

```diff
@@ -1,184 +1,185 @@
-        from models.models import create_model
-        sys.argv = [sys.argv[0]]
-        os.mkdir(self.input_file_grp+"/test_A/")
-        opt = TestOptions().parse(save=False)
-        opt.nThreads = 1   # test code only supports nThreads = 1
-        opt.batchSize = 1  # test code only supports batchSize = 1
-        opt.serial_batches = True  # no shuffle
-        opt.no_flip = True  # no flip
-        opt.rood_dir = self.input_file_grp # make into proper path
-        opt.checkpoints_dir = self.parameter['checkpoint_dir'] 
-        opt.dataroot = self.input_file_grp
-        opt.name = self.parameter['model_name']
-        opt.label_nc = 0
-        opt.no_instance = True
-        opt.resize_or_crop = self.parameter['imgresize']
-        opt.n_blocks_global = 10
-        opt.n_local_enhancers = 2
-        opt.gpu_ids = [self.parameter['gpu_id']]
-        opt.loadSize = self.parameter['resizeHeight']
-        opt.fineSize = self.parameter['resizeWidth']
-        
-        model = create_model(opt)
-    
-        return opt, model
-
-    def prepare_data(self, opt, page_img, path):
-
-        sys.path.append(path)
-        from data.data_loader import CreateDataLoader
-        
-        data_loader = CreateDataLoader(opt)
-        data_loader.dataset.A_paths = [page_img.filename]
-        data_loader.dataset.dataset_size = len(data_loader.dataset.A_paths)
-        data_loader.dataloader = torch.utils.data.DataLoader(data_loader.dataset,
-                                                             batch_size=opt.batchSize,
-                                                             shuffle=not opt.serial_batches,
-                                                             num_workers=int(opt.nThreads))
-        dataset = data_loader.load_data()
-        return dataset   
-        # test
-        
-    def process(self):
-        try:
-            page_grp, self.image_grp = self.output_file_grp.split(',')
-        except ValueError:
-            page_grp = self.output_file_grp
-            self.image_grp = FALLBACK_IMAGE_GRP
-            LOG.info("No output file group for images specified, falling back to '%s'", FALLBACK_IMAGE_GRP)
-        if not torch.cuda.is_available():
-            LOG.warning("torch cannot detect CUDA installation.")
-            self.parameter['gpu_id'] = 'cpu'
 
-        path = self.parameter['pix2pixHD']
-        if not Path(path).is_dir():
-            LOG.error("""\
-                NVIDIA's pix2pixHD was not found at '%s'. Make sure the `pix2pixHD` parameter 
-                in ocrd-tools.json points to the local path to the cloned pix2pixHD repository.
+    def process(self):
 
-                pix2pixHD can be downloaded from https://github.com/NVIDIA/pix2pixHD
-                """ % path)
+        if len(self.output_file_grp.split(',')) > 1:
+            LOG.error("Expects exactly one output file group, not %s", len(self.output_file_grp.split(',')))
             sys.exit(1)
```

Comment:
Just curious: Is this a recipe you would recommend for other processors, too?

Reply:
Yes, not sure whether the error message is ideal though. Maybe explictly state that commas delimit groups?

Reply:
Yes. Moreover, the `%s` does not fit the number context. And I would personally prefer getting an exception here:
```suggestion
        assert len(self.output_file_grp.split(',')) == 1, \
        "Expected exactly one output file group, but '%s' has %d" % (
            self.output_file_grp, len(self.output_file_grp.split(',')))
```

Reply:
54688e7